obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    physical measured information).
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+import warnings
+
+import numpy as np
+import os
+
+from obspy import UTCDateTime
+from obspy.core.event import Event, Origin, Magnitude, Comment, Catalog
+from obspy.core.event import EventDescription, CreationInfo
+from obspy.core.event import Pick, WaveformStreamID, Arrival, Amplitude
+
+
+def is_sfile(sfile):
+    """
+    Basic test of whether the file is nordic format or not.
+
+    Not exhaustive, but checks some of the basics.
+
+    :type sfile: str
+    :param sfile: Path to sfile
+    :rtype: bool
+    """
+    if not hasattr(sfile, "readline"):
+        try:
+            with open(sfile, 'r') as f:
+                topline = f.readline()
+                if not len(topline.rstrip()) == 80:
+                    return False
+                f.seek(0)
+                for line in f:
+                    if line[79] in [' ', '1']:
+                        head_line = line
+                        break
+        except:
+            return False
+    else:
+        topline = sfile.readline()
+        if not len(topline.rstrip()) == 80:
+            return False
+        sfile.seek(0)
+        for line in sfile:
+            if line[79] in [' ', '1']:
+                head_line = line
+                break
+    if 'head_line' in locals():
+        try:
+            sfile_seconds = int(head_line[16:18])
+        except ValueError:
+            return False
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        try:
+            UTCDateTime(int(head_line[1:5]), int(head_line[6:8]),
+                        int(head_line[8:10]), int(head_line[11:13]),
+                        int(head_line[13:15]), sfile_seconds,
+                        int(head_line[19:20]) * 100000) + add_seconds
+            return True
+        except:
+            return False
+    else:
+        return False
+
+
+def _int_conv(string):
+    """
+    Convenience tool to convert from string to integer.
+    If empty string return a 999 rather than an error.
+
+    >>> _int_conv('12')
+    12
+    >>> _int_conv('')
+    999
+    """
+    try:
+        intstring = int(string)
+    except:
+        intstring = 999
+    return intstring
+
+
+def _float_conv(string):
+    """
+    Convenience tool to convert from string to float.
+    If empty string return NaN rather than an error.
+
+    >>> _float_conv('12')
+    12.0
+    >>> _float_conv('')
+    999.0
+    >>> _float_conv('12.324')
+    12.324
+    """
+    try:
+        floatstring = float(string)
+    except:
+        # cjh Changed this to deal with QuakeML validation issues with 'NaN'
+        floatstring = float('999')
+    return floatstring
+
+
+def _str_conv(number, rounded=False):
```

Comment:
OK, I was ignoring the `999` sections, but if they're still relevant with `None`, then maybe this function is useful there. I still think it should _only_ be used where values are optional, otherwise things "work" but are really invalid.